src/overlays/gamestates/ovl_file_choose/file_choose.h

```diff
@@ -0,0 +1,219 @@
+#ifndef _FILE_CHOOSE_H_
+#define _FILE_CHOOSE_H_
+
+#include "ultra64.h"
+#include "global.h"
+#include "vt.h"
+
+
+#define GET_NEWF(sramCtx, slotNum, index) (sramCtx->readBuff[gSramSlotOffsets[slotNum] + OFFSETOF(SaveContext, newf[index])])
+
+#define SLOT_OCCUPIED(sramCtx, slotNum) \
+    ((GET_NEWF(sramCtx, slotNum, 0) == 'Z') || \
+     (GET_NEWF(sramCtx, slotNum, 1) == 'E') || \
+     (GET_NEWF(sramCtx, slotNum, 2) == 'L') || \
+     (GET_NEWF(sramCtx, slotNum, 3) == 'D') || \
+     (GET_NEWF(sramCtx, slotNum, 4) == 'A') || \
+     (GET_NEWF(sramCtx, slotNum, 5) == 'Z'))
+
+// Init mode: Initial setup as the file select is starting up, fades and slides in various menu elements
+// Config mode: Handles the bulk of the file select, various configuration tasks like picking a file, copy/erase, and the options menu
+// Select mode: Displays the selected file with various details about it, and allows the player to confirm and open it
+typedef enum {
+    /* 0 */ MENU_MODE_INIT,
+    /* 1 */ MENU_MODE_CONFIG,
+    /* 2 */ MENU_MODE_SELECT
+} MenuMode;
+
+typedef enum {
+    /* 00 */ CM_FADE_IN_START,
+    /* 01 */ CM_FADE_IN_END,
+    /* 02 */ CM_MAIN_MENU,
+    /* 03 */ CM_SETUP_COPY_SOURCE,
+    /* 04 */ CM_SELECT_COPY_SOURCE,
+    /* 05 */ CM_SETUP_COPY_DEST_1,
+    /* 06 */ CM_SETUP_COPY_DEST_2,
+    /* 07 */ CM_SELECT_COPY_DEST,
+    /* 08 */ CM_EXIT_TO_COPY_SOURCE_1,
+    /* 09 */ CM_EXIT_TO_COPY_SOURCE_2,
+    /* 10 */ CM_SETUP_COPY_CONFIRM_1,
+    /* 11 */ CM_SETUP_COPY_CONFIRM_2,
+    /* 12 */ CM_COPY_CONFIRM,
+    /* 13 */ CM_RETURN_TO_COPY_DEST,
+    /* 14 */ CM_COPY_ANIM_1,
+    /* 15 */ CM_COPY_ANIM_2,
+    /* 16 */ CM_COPY_ANIM_3,
+    /* 17 */ CM_COPY_ANIM_4,
+    /* 18 */ CM_COPY_ANIM_5,
+    /* 19 */ CM_COPY_RETURN_MAIN,
+    /* 20 */ CM_SETUP_ERASE_SELECT,
+    /* 21 */ CM_ERASE_SELECT,
+    /* 22 */ CM_SETUP_ERASE_CONFIRM_1,
+    /* 23 */ CM_SETUP_ERASE_CONFIRM_2,
+    /* 24 */ CM_ERASE_CONFIRM,
+    /* 25 */ CM_EXIT_TO_ERASE_SELECT_1,
+    /* 26 */ CM_EXIT_TO_ERASE_SELECT_2,
+    /* 27 */ CM_ERASE_ANIM_1,
+    /* 28 */ CM_ERASE_ANIM_2,
+    /* 29 */ CM_ERASE_ANIM_3,
+    /* 30 */ CM_EXIT_ERASE_TO_MAIN,
+    /* 31 */ CM_UNUSED_31,
+    /* 32 */ CM_ROTATE_TO_NAME_ENTRY,
+    /* 33 */ CM_NAME_ENTRY,
+    /* 34 */ CM_START_NAME_ENTRY,
+    /* 35 */ CM_NAME_ENTRY_TO_MAIN,
+    /* 36 */ CM_MAIN_TO_OPTIONS,
+    /* 37 */ CM_OPTIONS_MENU,
+    /* 38 */ CM_START_OPTIONS,
+    /* 39 */ CM_OPTIONS_TO_MAIN,
+    /* 40 */ CM_UNUSED_DELAY
+} ConfigMode;
+
+typedef enum {
+    /* 00 */ SM_FADE_MAIN_TO_SELECT,
+    /* 01 */ SM_MOVE_FILE_TO_TOP,
+    /* 02 */ SM_FADE_IN_FILE_INFO,
+    /* 03 */ SM_CONFIRM_FILE,
+    /* 04 */ SM_FADE_OUT_FILE_INFO,
+    /* 05 */ SM_MOVE_FILE_TO_SLOT,
+    /* 06 */ SM_FADE_OUT,
+    /* 07 */ SM_LOAD_GAME
```

Comment:
```suggestion
    /* 0 */ SM_FADE_MAIN_TO_SELECT,
    /* 1 */ SM_MOVE_FILE_TO_TOP,
    /* 2 */ SM_FADE_IN_FILE_INFO,
    /* 3 */ SM_CONFIRM_FILE,
    /* 4 */ SM_FADE_OUT_FILE_INFO,
    /* 5 */ SM_MOVE_FILE_TO_SLOT,
    /* 6 */ SM_FADE_OUT,
    /* 7 */ SM_LOAD_GAME
```
Remove leading zeros